base/src/main/java/proguard/optimize/inline/lambdalocator/LambdaLocator.java

```diff
@@ -0,0 +1,135 @@
+package proguard.optimize.inline.lambdalocator;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import proguard.backport.LambdaExpression;
+import proguard.backport.LambdaExpressionCollector;
+import proguard.classfile.ClassPool;
+import proguard.classfile.Clazz;
+import proguard.classfile.Method;
+import proguard.classfile.ProgramClass;
+import proguard.classfile.ProgramMethod;
+import proguard.classfile.attribute.CodeAttribute;
+import proguard.classfile.attribute.visitor.AllAttributeVisitor;
+import proguard.classfile.constant.ClassConstant;
+import proguard.classfile.constant.FieldrefConstant;
+import proguard.classfile.constant.Utf8Constant;
+import proguard.classfile.constant.visitor.ConstantVisitor;
+import proguard.classfile.instruction.ConstantInstruction;
+import proguard.classfile.instruction.Instruction;
+import proguard.classfile.instruction.visitor.AllInstructionVisitor;
+import proguard.classfile.instruction.visitor.InstructionVisitor;
+import proguard.classfile.visitor.MemberVisitor;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * A class that in a program locates the starting point of a potential Kotlin lambda (sometimes called static lambdas
```

Comment:
`sometimes called static lambdas in the code` -> Can you make sure that you refer to 1 concept with 1 name? I think 'kotlin lambdas' would make the most sense, as that captures best what you're covering?